ext/session/session.c

```diff
@@ -1335,8 +1335,8 @@ static zend_result php_session_send_cookie(void) /* {{{ */
 	}
 
 	/* Prevent broken Set-Cookie header, because the session_name might be user supplied */
-	if (strpbrk(PS(session_name), SESSION_FORBIDDEN_CHARS) != NULL) {   /* man isspace for \013 and \014 */
-		php_error_docref(NULL, E_WARNING, "session.name cannot contain any of the following '=,;.[ \\t\\r\\n\\013\\014'");
+	if (strpbrk(PS(session_name), SESSION_FORBIDDEN_CHARS) != NULL || memchr(PS(session_name), '\0', strlen(PS(session_name))) != NULL) {   /* man isspace for \013 and \014 */
+		php_error_docref(NULL, E_WARNING, "session.name cannot contain any of the following '=,;.[ \\t\\r\\n\\013\\014\\0'");
 		return FAILURE;
 	}
```

Comment:
In what other situations would there be a nul byte in the session name.
Moreover, you are not going to find any nul bytes if you use `strlen()` for the length as it will stop at the first nul byte.

Reply:
I removed this part. We can set the session name through `ini_set()`, but you are right this code doesn't detect it.